### PR TITLE
FEATURE: allow tags from tag field to be confined to a tag group

### PR DESCRIPTION
**Pull request by fzngagan**

### CODE REVIEW

File: extensions/discourse_tagging.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module ::DiscourseTagging
+  class << self
+    alias_method :discourse_core_filter_allowed_tags, :filter_allowed_tags
```
Comment (angusmcleod) [inline extensions/discourse_tagging.rb:5]: @fzngagan Any reason we're not using a module prepension and ``super`` here?

File: extensions/discourse_tagging.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    def filter_allowed_tags(guardian, opts = {})
+      if tagGroups = RequestStore.store[:tagGroups]
```
Comment (angusmcleod) [inline extensions/discourse_tagging.rb:7]: @fzngagan We always use ``underscore_case`` for variables in ruby.

**Review by angusmcleod — CHANGES_REQUESTED** [review]
@fzngagan A few minor formatting changes, and an rspec and this will be ready.

File: extensions/discourse_tagging.rb (hunk 1 above)
Comment (fzngagan) [reply extensions/discourse_tagging.rb:5]: I thought it wasn't possible but did some reading and experiments and yes, there's a way to do that. Will update the code accordingly.

File: extensions/discourse_tagging.rb (hunk 2 above)
Comment (fzngagan) [reply extensions/discourse_tagging.rb:7]: Noted.

Comment (fzngagan) [reply extensions/discourse_tagging.rb]: Fixed.

File: extensions/discourse_tagging.rb (hunk 1 above)
Comment (fzngagan) [reply extensions/discourse_tagging.rb]: Fixed

**Comment by angusmcleod** [comment]
@fzngagan let me know when I can merge

**Comment by fzngagan** [comment]
> @fzngagan let me know when I can merge

Hey, I'm just doing a quick test on stable right now.

**Comment by fzngagan** [comment]
@angusmcleod just tested. Works fine on latest stable without any changes. :)

**Pull request merged**